### ISSUE TITLE
[Bug]: Community moderators reordered by user ID after 0.18.0 update

### ISSUE THREAD

**Issue by xaegreal** [LemmyNet/lemmy#3382]
### Requirements

- [X] Is this a bug report? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a single bug? Do not put multiple bugs in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Summary

When a new user creates a community and appoints an older user as mod to it, older mod is listed before the new one. Also I can't transfer it back.

Note: I thought the migrations were working at the start of the lemmy. For this reason, I did the update only by pulling the current version of ansible without any extra action. Was it necessary to migrate? I couldn't find an article about it.

### Steps to Reproduce

1. Create two users.
2. Create a community with newer user.
3. Appoint older user as mod to the community.

Mods listed in _older user_, _newer user_ order. But it should have been reversed because newer user created the community.

### Technical Details

compose logs. no output from lemmy.
`proxy_1     | 192.168.16.1 - - [27/Jun/2023:19:41:46 +0000] "POST /api/v3/community/transfer HTTP/1.1" 200 1563 "https://lemmynsfw.com/post/116863" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/114.0"`

_/api/v3/community/transfer_ endpoint response:
```json
{
  "community_view": {
    "community": {
      "id": 4943,
      "name": "test3",
      "title": "test3",
      "removed": false,
      "published": "2023-06-27T19:34:19.806342",
      "deleted": false,
      "nsfw": false,
      "actor_id": "https://lemmynsfw.com/c/test3",
      "local": true,
      "hidden": false,
      "posting_restricted_to_mods": false,
      "instance_id": 1
    },
    "subscribed": "NotSubscribed",
    "blocked": false,
    "counts": {
      "id": 1755,
      "community_id": 4943,
      "subscribers": 1,
      "posts": 1,
      "comments": 1,
      "published": "2023-06-27T19:34:19.806342",
      "users_active_day": 2,
      "users_active_week": 2,
      "users_active_month": 2,
      "users_active_half_year": 2,
      "hot_rank": 1569
    }
  },
  "moderators": [
    {
      "community": {
        "id": 4943,
        "name": "test3",
        "title": "test3",
        "removed": false,
        "published": "2023-06-27T19:34:19.806342",
        "deleted": false,
        "nsfw": false,
        "actor_id": "https://lemmynsfw.com/c/test3",
        "local": true,
        "hidden": false,
        "posting_restricted_to_mods": false,
        "instance_id": 1
      },
      "moderator": {
        "id": 7776,
        "name": "user",
        "banned": false,
        "published": "2023-06-12T08:25:13.926052",
        "actor_id": "https://lemmynsfw.com/u/user",
        "local": true,
        "deleted": false,
        "admin": false,
        "bot_account": false,
        "instance_id": 1
      }
    },
    {
      "community": {
        "id": 4943,
        "name": "test3",
        "title": "test3",
        "removed": false,
        "published": "2023-06-27T19:34:19.806342",
        "deleted": false,
        "nsfw": false,
        "actor_id": "https://lemmynsfw.com/c/test3",
        "local": true,
        "hidden": false,
        "posting_restricted_to_mods": false,
        "instance_id": 1
      },
      "moderator": {
        "id": 236581,
        "name": "user2",
        "banned": false,
        "published": "2023-06-27T19:33:02.332929",
        "actor_id": "https://lemmynsfw.com/u/user2",
        "local": true,
        "deleted": false,
        "admin": false,
        "bot_account": false,
        "instance_id": 1
      }
    }
  ],
  "discussion_languages": []
}
```

### Version

0.18.0

### Lemmy Instance URL

lemmynsfw.com

**Comment by xaegreal** [comment]
Sorry for mentioning, but I believe this is a major issue. @Nutomic @dessalines

**Comment by snowe2010** [comment]
does moderator order matter in Lemmy? Is the 'top mod' not just the one that created it? Or does it follow reddit rules?

**Comment by xaegreal** [comment]
> does moderator order matter in Lemmy? Is the 'top mod' not just the one that created it? Or does it follow reddit rules?

IDK how things working on Reddit but top mod is the one able to add/remove mods to the community.

**Comment by snowe2010** [comment]
@xaegreal any mod is currently able to add mods (not sure about remove).

**Comment by xaegreal** [comment]
It fixed with the 0.18.1 update.
